packages/block-library/src/heading/edit.js

```diff
@@ -24,9 +24,13 @@ import {
 import { Platform } from '@wordpress/element';
 
 function HeadingEdit( { attributes, setAttributes, mergeBlocks, onReplace } ) {
-	const { align, content, level, placeholder } = attributes;
+	const { align, content, level, placeholder, style } = attributes;
 	const tagName = 'h' + level;
 
+	const styles = {
+		color: style && style.color && style.color.text
+	};
```

Comment:
Can we do this in a more generic way, in a hook or something because this will apply to all blocks with that flag enabled?

Reply:
I am not sure we know how to do that. Could you maybe explain with some sample code?

Reply:
@youknowriad can you help us with that?

Reply:
Right now the "style" prop is injected into the Block Wrapper using getEditWrapperProps in the hook for the web. Can we do the same for mobile to inject `styles`?

Reply:
https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/hooks/color.js#L110-L116

The difference here would be that we should run the "save" logic in the "edit" one because the props will be different on mobile.

Reply:
That link was for the color hook, here's the one for the style one https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/hooks/style.js#L118-L125

Reply:
@youknowriad I don't think we are in a position in native to do this like that, we will need to change our components to always read from styles props and this is not happening at the moment. I think for now we should merge this PR as it is to solve our breakages and then try to do follow up PRs on updating our native code to better support this generic styles in a more generic way.

Reply:
I'm not suggesting it should be solved now but mking `getEditWrapperProps` work on mobile is key yes.